utils/dataset_utils.py

```diff
@@ -68,6 +72,67 @@
 
 _MAX_ROWS = 200000
 
+def initialize_cache_hub_repo(cache_path, dataset_cache_dir):
+    """
+    This function tries to initialize a dataset cache on the huggingface hub. The
+    function expects you to have HUB_CACHE_ORGANIZATION=<the organization you've set up on the hub to store your cache>
+    and HF_TOKEN=<your hf token> on separate lines in a file named .env at the root of this repo.
+
+    Args:
+        cache_path (string):
+            The path to the local dataset cache.
+        dataset_cache_dir (string):
+            The name of the dataset repo on the huggingface hub that you want.
+    """
+
+    if Path(".env").is_file():
+        load_dotenv(".env")
+    HF_TOKEN = getenv("HF_TOKEN")
+    HUB_CACHE_ORGANIZATION = getenv("HUB_CACHE_ORGANIZATION")
+    repo = Repository(local_dir=cache_path,
+                      clone_from=HUB_CACHE_ORGANIZATION + "/" + dataset_cache_dir,
+                      repo_type="dataset", use_auth_token=HF_TOKEN)
+    repo.lfs_track(["*.feather"])
+    return repo
+
+def pull_cache_from_hub(cache_path, dataset_cache_dir):
+    """
+    This function tries to pull a datasets cache from the huggingface hub if a
+    cache for the dataset does not already exist locally. The function expects you
+    to have you HUB_CACHE_ORGANIZATION=<the organization you've set up on the hub to store your cache>
+    and HF_TOKEN=<your hf token> on separate lines in a file named .env at the root of this repo.
+
+    Args:
+        cache_path (string):
+            The path to the local dataset cache that you want.
+        dataset_cache_dir (string):
+            The name of the dataset repo on the huggingface hub.
+
+    Returns:
+        string: a log about whether the cache was pulled or not
+    """
+
+    if Path(".env").is_file():
+        load_dotenv(".env")
+
+    HF_TOKEN = getenv("HF_TOKEN")
+    HUB_CACHE_ORGANIZATION = getenv("HUB_CACHE_ORGANIZATION")
+
+    log = "Pulled cache from hub!"
+    if not isdir(cache_path):
+        if dataset_cache_dir in [
+            dataset_info.id.split("/")[-1] for dataset_info in
```

Comment:
Can you add a comment about what dataset_info.id is, and why -1 would be the choice for selecting from it? I want to have a better sense of what this is doing when I read it.